packages/server/lib/controllers/sync.controller.ts

```diff
@@ -118,14 +118,33 @@ class SyncController {
                 return;
             }
 
-            const syncs = await getSyncs(connection, orchestrator);
+            const rawSyncs = await getSyncs(connection, orchestrator);
+
+            const syncs = await this.addObjectCount(rawSyncs, connection.id!, environment.id);
 
             res.send(syncs);
         } catch (e) {
             next(e);
         }
     }
 
+    private async addObjectCount(syncs: (Sync & { models: string[] })[], connectionId: number, environmentId: number) {
+        const objectCountByModelResult = await recordsService.getRecordCountsByModel({ connectionId, environmentId });
+
+        if (objectCountByModelResult.isOk()) {
+            const objectCountByModel = objectCountByModelResult.unwrap();
+            return syncs.map((sync) => {
+                let objectCountSum = 0;
+                for (const model of sync.models) {
+                    objectCountSum = objectCountSum + (objectCountByModel[model]?.object_count ?? 0);
+                }
+                return { ...sync, object_count: objectCountSum };
+            });
```

Comment:
can we use reduce instead of the for loop
```
  return syncs.map(sync => ({
    ...sync,
    object_count: sync.models.reduce(
      (sum, model) => sum + (result.value[model]?.object_count ?? 0),
      0
    )
  }));
```